### PR TITLE
fix(secu): check ACL rights when sending externalcommands

### PR DESCRIPTION
**Pull request by adr-mo**
This PR aims to control what the user can send in centcore pipe using the API. It will check that user can (_or not_) submit a specific command regarding its ACLs

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x
- [x] 19.10.x

<h2> How this pull request can be tested ? </h2>

Procedure is described in the JIRA ticket.

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/api/class/centreon_monitoring_externalcmd.class.php
Comment (sc979) [inline www/api/class/centreon_monitoring_externalcmd.class.php:102]:
```suggestion
                $userAcl = new CentreonACL($centreon->user->user_id, $isAdmin);
```

File: www/api/class/centreon_monitoring_externalcmd.class.php
Comment (sc979) [inline www/api/class/centreon_monitoring_externalcmd.class.php:115]: why declare $isAdmin if you don't use it ?
https://github.com/centreon/centreon/pull/9366/files#diff-5b28191bbed53758319095816ed02e7311c3120918388e11c7f1b0e86899af9aR96

File: www/api/class/centreon_monitoring_externalcmd.class.php
Comment (sc979) [inline www/api/class/centreon_monitoring_externalcmd.class.php:91]: are you that this is not needed elsewhere ?

File: www/api/class/centreon_monitoring_externalcmd.class.php
Comment (kduret) [inline www/api/class/centreon_monitoring_externalcmd.class.php:115]:
```suggestion
                    if (!$isAdmin) {
```

File: www/api/class/centreon_monitoring_externalcmd.class.php
Comment (kduret) [inline www/api/class/centreon_monitoring_externalcmd.class.php:37]: prefer adding `/` in case of trailing slash is missing in `_CENTREON_PATH_`
```suggestion
require_once _CENTREON_PATH_ . '/www/class/centreonExternalCommand.class.php';
```

File: www/api/class/centreon_monitoring_externalcmd.class.php
Comment (kduret) [reply www/api/class/centreon_monitoring_externalcmd.class.php:91]: it's ok cause `if` return this array, and `else` throw an exception

**Review by kduret — COMMENTED** [review]
LGTM once suggestions are merged

**Review by sc979 — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**
